test/GasAccounting.t.sol

```diff
@@ -99,307 +124,515 @@ contract MockGasAccounting is GasAccounting, Test {
+        tearDown();
+    }
 
-        mockGasAccounting.setPhase(ExecutionPhase.SolverOperation);
-        mockGasAccounting.setSolverLock(solverOp.from);
+    function test_reconcile_invalidAccess_reverts() public {
+        // Set phase to SolverOperation and set solver lock
+        mockGasAccounting.setLock(executionEnvironment, 0, uint8(ExecutionPhase.SolverOperation));
+        mockGasAccounting.setSolverTo(solverOneEOA);
 
-        assertTrue(mockGasAccounting.reconcile(0) > 0);
+        // Ensure bonded balance is set
+        mockGasAccounting.increaseBondedBalance(solverOneEOA, 10 ether);
+
+        // Expect revert if called by the wrong address
+        vm.expectRevert(AtlasErrors.InvalidAccess.selector);
+        mockGasAccounting.reconcile(0);
+        tearDown();
+    }
+
+    function test_reconcile_withCorrectAddress() public {
+        // Set initial claims and deposits
+        mockGasAccounting.setClaims(20 ether); // Increased claims to ensure deductions are higher
+        mockGasAccounting.setLock(executionEnvironment, 0, uint8(ExecutionPhase.SolverOperation));
+        deal(executionEnvironment, 10 ether);
+        vm.prank(executionEnvironment);
+        mockGasAccounting.contribute{ value: 10 ether }();
+
+        mockGasAccounting.setSolverTo(executionEnvironment);
+
+        // Ensure bonded balance is set
+        mockGasAccounting.increaseBondedBalance(executionEnvironment, 10 ether);
+
+        // Call reconcile with the correct execution environment
+        vm.prank(executionEnvironment);
+        uint256 result = mockGasAccounting.reconcile(5 ether);
+        assertTrue(result > 0, "Reconcile should return a value greater than zero");
     }
 
     function test_reconcile() public {
-        // FIXME: fix before merging spearbit-reaudit branch
-        vm.skip(true);
+        // Ensure the execution environment starts with zero balance
+        deal(executionEnvironment, 0);
+
+        // Fund the contract to allow reconciliation
+        deal(executionEnvironment, initialClaims);
 
+        // Set phase to SolverOperation and set solver lock
         mockGasAccounting.setPhase(ExecutionPhase.SolverOperation);
-        mockGasAccounting.setSolverLock(solverOp.from);
-        assertTrue(mockGasAccounting.reconcile{ value: initialClaims }(0) == 0);
+        mockGasAccounting.setSolverLock(solverOneEOA);
+        mockGasAccounting.setSolverTo(solverOneEOA);
+
+        // Call reconcile with the correct execution environment
+        vm.prank(solverOneEOA);
+        assertTrue(mockGasAccounting.reconcile{ value: initialClaims }(0) == 0, "Reconcile should return zero");
+
+        // Verify solver lock data
         (address currentSolver, bool verified, bool fulfilled) = mockGasAccounting.solverLockData();
-        assertTrue(verified && fulfilled);
-        assertEq(currentSolver, solverOp.from);
+        assertTrue(verified && fulfilled, "Solver should be verified and fulfilled");
```

Comment:
Could break this up into 2 assertTrue lines, just to be clearer where things are breaking if we change code

Reply:
done